forem/forem
# Add Options to "Invalidate" and "Mark as Valid" Vomit Reactions

## What type of PR is this? (check all applicable)


## Description
This PR adds two buttons, "Invalidate" and "Mark as Valid," to `_abuse_reports.html.erb` that gives Admins the option to un-confirm a confirmed vomit reaction and reopen an invalidated vomit reaction. To accomplish this, this PR adds a conditional button to `_abuse_reports.html.erb` and a couple of functions to handle the un-confirming and reopening of a vomit reaction in `reaction_controller.js`.

## Related Tickets & Documents

## QA Instructions, Screenshots, Recordings
- To QA these changes, ensure that you are a `super_admin` and navigate to `/admin/reports` in your browser:
![Screen Shot 2020-08-13 at 9 49 56 AM](https://user-images.githubusercontent.com/32834804/90156962-5c0e9780-dd4a-11ea-9a15-1f083d4686a6.png)

- Once you have navigated there, click "confirm" on one of the vomit reactions located under "Vomit Reactions" -> "Open/Unresolved":
![Screen Shot 2020-08-13 at 9 51 18 AM](https://user-images.githubusercontent.com/32834804/90157106-8a8c7280-dd4a-11ea-9edc-2bd36b7a0c64.png)

- The confirmed vomit can then be found under the "Resolved" tab:
![Screen Shot 2020-08-14 at 2 50 30 PM](https://user-images.githubusercontent.com/32834804/90291656-9573ff80-de3d-11ea-8458-5b54c17ee3bf.png)


- Click "Invalidate" on the confirmed vomit reaction. This will "invalidate" the vomit, placing it under the "Invalid" tab:
<img width="1197" alt="Screen Shot 2020-08-17 at 2 15 51 PM" src="https://user-images.githubusercontent.com/32834804/90440558-a7031480-e094-11ea-8a5f-0ce5c6188eac.png">


- From there, go to the "Invalid" tab and click "Mark as Valid". This will place the vomit reaction previously marked as "invalid" underneath the "Resolved" tab, providing an Admin with more flexibility in case something is wrongly marked:
<img width="1197" alt="Screen Shot 2020-08-17 at 2 21 11 PM" src="https://user-images.githubusercontent.com/32834804/90440720-f5b0ae80-e094-11ea-8802-c12e3968bba9.png">

- Try to break things! ⚒️ 
 
## Added tests?


## Added to documentation?


## [optional] Are there any post-deployment tasks we need to perform?
N/A

## [optional] What gif best describes this PR or how it makes you feel?

![erase](https://media.giphy.com/media/WS6CDvvyNDrhZRFBtT/giphy.gif)
